src/classes/UTIL_Describe.cls

```diff
@@ -370,4 +370,108 @@ public class UTIL_Describe {
+     */
+    public static Map<String, Id> getRecTypesMapByDevName(String objectName) {	    
+	    return getRecTypesMap(objectName, true);
+	}
+	
+	/*******************************************************************************************************
+     * @description Returns a map of active, user-available RecordType IDs for a given SObjectType,
+     * keyed by each RecordType's Name. 
+     * @param objectName Name of the object to retrieve record types for.
+     * @return Map<String, Id> Map of Name-to-ID of available record types for object specified. 
+     */
+    @AuraEnabled
+    public static Map<String, Id> getRecTypesMapByName(String objectName) {
+        return getRecTypesMap(objectName, false);
+    }
+    
+    /*******************************************************************************************************
+     * @description Returns a map of active, user-available RecordType IDs for a given SObjectType, keyed
+     * by either the RecordType's Name or the Developer Name, depending on the value of the second param. 
+     * @param objectName Name of the object to retrieve record types for.
+     * @param devName True if we want the key of the returned map to be the record type's developer name.
+     * False if we want it to be the friendly name.
+     * @return Map<String, Id> Map of Name-to-ID or DeveloperName-to-ID (depending on the second paramater), 
+     * of available record types for object specified. 
+     */
+    private static Map<String, Id> getRecTypesMap(String objectName, Boolean devName) {      
+        if(mapRecordTypes == null) {
+            mapRecordTypes = new Map<String, Id>();
+        
+            // Get the Describe Result
+            fillMapsForObject(objectName);
+            Schema.DescribeSObjectResult obj = objectDescribes.get(objectName);
+        
+            // Obtain ALL Active Record Types for the given SObjectType objectType
+            String soql = 'SELECT Id, Name, DeveloperName FROM RecordType ' + 'WHERE SObjectType = \'' 
+                          + String.escapeSingleQuotes(obj.getName()) + '\' ' + 'AND IsActive = TRUE';
+            List<SObject> results;
+            try {
+                results = Database.query(soql);
+            } catch (Exception ex) {
+                results = new List<SObject>();
+            }
+        
+            // Obtain the RecordTypeInfos for this SObjectType objectType
+            Map<Id,Schema.RecordTypeInfo> recordTypeInfos = obj.getRecordTypeInfosByID();
+        
+            for (SObject rt : results) { 
+                // Check RecordType IS available for the running user
+                if (!Test.isRunningTest()) {
+                    if (recordTypeInfos.get(rt.Id).isAvailable()) {
+                        if(devName) {
+                            mapRecordTypes.put(String.valueOf(rt.get('DeveloperName')), rt.Id);
```

Comment:
Sorry, i'm not sure i'm following this - why are you pulling 'DeveloperName' here?